### PR TITLE
Login design

### PR DESCRIPTION
**Pull request by himanshm**
Initial Design of Login Page
Add design to the login page
Change auth store to compositon api
import vuetify library and its components
Change directory name of TheLogin.vue file

### CODE REVIEW

File: src/views/auth/TheLogin.vue [hunk 1]
```diff
@@ -0,0 +1,154 @@
+<script setup lang="ts">
+// import { storeToRefs } from 'pinia'
+import { useAuthStore } from '../../stores/auth.ts'
+import { ref } from 'vue'
+import { useRouter } from 'vue-router'
+
+import loginImg from '../../assets/identity.svg'
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:7]: Why don't we create an svg folder img inside assets and keep all the svgs over there.

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/assets/identity.svg
Comment (AishwaryShrivastav) [inline src/assets/identity.svg:1]: rename the file with proper name

File: src/stores/auth.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  return {
+    userId,
+    authToken,
+    userLogin,
+    isAuthenticated
+  }
+}
+
+// export const useAuthStore = defineStore({
```
Comment (AishwaryShrivastav) [inline src/stores/auth.ts:38]: remove commented code if its not for future use

File: src/views/auth/TheLogin.vue [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+const username = ref('')
+const password = ref('')
+const showPassword = ref(false)
+
+const login = async () => {
+  await userLogin(username.value, password.value)
+  const isAutheticated = authStore.isAuthenticated
+  if (isAutheticated) {
+    // Redirect to home page
+    // Use replace to not enable the user to return to login route
```
[the remark this answers is not in the document]
Comment (himanshm) [reply src/views/auth/TheLogin.vue:21]: Implementing this in next PR

File: src/stores/auth.ts [hunk 4]
```diff
@@ -1,43 +1,36 @@
...
-    }
-    // logout() {
-    //   this.isLoggedIn = false
-    //   this.authToken = null
-    // }
-  },
-  getters: {
-    getLoginStatus(state) {
-      return state.isLoggedIn
+export function useAuthStore() {
```
Comment (disha1202) [inline src/stores/auth.ts:8]: Use defineStore

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/stores/auth.ts (hunk 4 above)
Comment (himanshm) [reply src/stores/auth.ts]: Done!

File: src/App.vue [hunk 5]
```diff
@@ -4,17 +4,30 @@ import BottomNavigation from './components/BottomNavigation.vue'
 import SideBar from './components/SideBar.vue'
 import Header from './components/TheHeader.vue'
 import { isMobile } from './utilities/helper'
+import { useRoute } from 'vue-router'
+import { computed } from 'vue'
+
+const route = useRoute()
+const isHomePage = computed(() => route.path === '/home')
```
Comment (disha1202) [inline src/App.vue:11]: We need to display side bar and bottom navigation on all the pages except for login. 
Not just on home page.

File: src/assets/images/logo.svg
Comment (disha1202) [inline src/assets/images/logo.svg:1]: Remove this file

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/stores/auth.ts [hunk 6]
```diff
@@ -1,43 +1,75 @@
...
+import type { Router } from 'vue-router'
+
+const baseUrl = import.meta.env.VITE_BASE_URL
+const url = `${baseUrl}/oauth2/access_token`
+
+export const useAuthStore = defineStore('auth', () => {
+  const authToken = useStorage('authToken', '')
+  const userId = useStorage('userId', '')
+  const tokenExpiry = useStorage('tokenExpirey', 0)
+  const didAutoLogout = ref(false)
```
Comment (disha1202) [inline src/stores/auth.ts:14]: Why do we need this variable?

File: src/stores/auth.ts [hunk 7]
```diff
@@ -1,43 +1,75 @@
...
+import { useStorage } from '@vueuse/core' // For state persistance
+import type { Router } from 'vue-router'
+
+const baseUrl = import.meta.env.VITE_BASE_URL
+const url = `${baseUrl}/oauth2/access_token`
+
+export const useAuthStore = defineStore('auth', () => {
+  const authToken = useStorage('authToken', '')
+  const userId = useStorage('userId', '')
+  const tokenExpiry = useStorage('tokenExpirey', 0)
```
Comment (disha1202) [inline src/stores/auth.ts:13]: Improve variable name

File: src/stores/auth.ts [hunk 8]
```diff
@@ -1,43 +1,75 @@
...
+          username,
+          password,
+          grant_type: 'password',
+          client_id: import.meta.env.VITE_CLIENT_ID
+        }
+      })
+      authToken.value = response.data.access_token
+      userId.value = username
+      const expiresIn = +response.data.expires_in * 1000 // Convert to milliseconds
+      // const expiresIn = 5000 // Testing
```
Comment (disha1202) [inline src/stores/auth.ts:33]: Remove commented code

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/router/index.ts [hunk 9]
```diff
@@ -10,14 +11,32 @@ const router = createRouter({
...
+  const authStore = useAuthStore()
+
+  if (to.meta.requiresAuth && !authStore.isAuthenticated()) {
+    next('/login')
+  } else if (to.meta.requiresUnauth && authStore.isAuthenticated()) {
+    next('/home')
+  } else {
+    next()
+  }
+})
```
Comment (disha1202) [inline src/router/index.ts:40]: Remove meta property and improve logic for routing guards

**Review by disha1202 — CHANGES_REQUESTED** [review]

**Comment by disha1202** [comment]
Resolve conflicts

File: src/stores/auth.ts [hunk 10]
```diff
@@ -1,43 +1,75 @@
...
+  }
+
+  const isAuthenticated = () => {
+    if (!authToken.value || !tokenExpiry.value) {
+      return false
     }
+    // check if the current time is passed token expiry time
+    // whether the current time is past the token's expiry time, which would mean that the token is no longer valid and the user is not authenticated.
+    return Date.now() < tokenExpiry.value
+  }
```
Comment (disha1202) [inline src/stores/auth.ts:50]:
```suggestion
   return !!authToken.value
  }
```

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/views/auth/TheLogin.vue [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+                      class="isax isax-eye pointer"
+                      v-if="showPassword"
+                      @click="showPassword = !showPassword"
+                    >
+                    </i>
+                    <i
+                      class="isax isax-eye-slash pointer"
+                      v-else
+                      @click="showPassword = !showPassword"
+                    ></i>
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:101]:
```suggestion
                    <i
                      :class=`isax ${showPassword ? 'isax-eye' : 'isax-eye-slash' } pointer`
                      v-if="showPassword"
                      @click="showPassword = !showPassword"
                    >
                    </i>
                    
```

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/stores/auth.ts (hunk 7 above)
Comment (himanshm) [reply src/stores/auth.ts:13]: changed it to validityDuration

File: src/views/auth/TheLogin.vue [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+            <h1 class="heading-text">{{ $t('OpenLlama') }}</h1>
+          </v-col>
+        </v-row>
+        <v-row>
+          <v-col>
+            <v-sheet max-width="600">
+              <p class="mb-2">{{ $t('EnterUsername') }}</p>
+              <v-form @submit.prevent="login">
+                <v-text-field
+                  class="input-fields"
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:68]: Remove this class

File: src/views/auth/TheLogin.vue [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+                      @click="showPassword = !showPassword"
+                    ></i>
+                  </template>
+                </v-text-field>
+                <div class="d-flex flex-row justify-center align-center">
+                  <v-checkbox
+                    :label="$t('RememberMe')"
+                    color="orange"
+                    hide-details
+                    class="checkbox"
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:109]: Remove this class

File: src/views/auth/TheLogin.vue [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+                  ></v-checkbox>
+                  <p class="forgot-password-btn">{{ $t('ForgotPassword') }}</p>
+                </div>
+                <p v-if="!loginIsValid">Please enter your username and password!</p>
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:113]: Why is this needed?

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/views/auth/TheLogin.vue (hunk 14 above)
Comment (himanshm) [reply src/views/auth/TheLogin.vue:113]: Removed!

File: src/views/auth/TheLogin.vue [hunk 15]
```diff
@@ -0,0 +1,165 @@
...
+const loginIsValid = ref(true)
+const snackbar = ref(false)
+
+const validateLogin = () => {
+  loginIsValid.value = true
+  if (username.value === '' && password.value === '') {
+    loginIsValid.value = false
+    snackbar.value = false
+  }
+}
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:23]: Remove this

File: src/views/auth/TheLogin.vue [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+
+const login = async () => {
+  validateLogin()
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:26]: Remove this

File: src/views/auth/TheLogin.vue [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+  if (!loginIsValid.value) {
+    return
+  }
+
+  await authStore.userLogin(username.value, password.value, router)
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:31]: Check if username and password is there, only then make api call.

File: src/views/auth/TheLogin.vue [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+
+  if (authStore.isAuthenticated()) {
+    router.replace({ name: 'home' })
+  } else {
+    console.error('Authentication Failed!')
+    snackbar.value = true // Show snackbar when authentication fails
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:37]: Improve variable name

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/views/auth/TheLogin.vue (hunk 13 above)
Comment (himanshm) [reply src/views/auth/TheLogin.vue:101]: Done!

File: src/views/auth/TheLogin.vue (hunk 15 above)
Comment (himanshm) [reply src/views/auth/TheLogin.vue:23]: Done!

File: src/views/auth/TheLogin.vue (hunk 16 above)
Comment (himanshm) [reply src/views/auth/TheLogin.vue:26]: Removed!

File: src/views/auth/TheLogin.vue (hunk 18 above)
Comment (himanshm) [reply src/views/auth/TheLogin.vue:37]: Made it showSnackbar

File: src/views/auth/TheLogin.vue [hunk 19]
```diff
@@ -0,0 +1,154 @@
...
+                >
+                  Login
+                </v-btn>
+              </v-form>
+            </v-sheet>
+          </v-col>
+        </v-row>
+      </v-col>
+    </v-row>
+    <div class="text-center">
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:103]: Remove this class

File: src/views/auth/TheLogin.vue [hunk 20]
```diff
@@ -0,0 +1,154 @@
...
+      <v-snackbar v-model="showSnackbar" color="var(--btn-color)">
+        Invalid Credentials
```
Comment (disha1202) [inline src/views/auth/TheLogin.vue:105]: Internationalize text

**Review by disha1202 — CHANGES_REQUESTED** [review]

**Review by disha1202 — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**